plotly/tests/test_core/test_plotly/test_plot.py

```diff
@@ -66,3 +66,99 @@ def test_plot_empty_data(self):
+                }
+            ]
+        }
+
+        kwargs = {'filename': 'invalid-sharing-argument',
+                  'sharing': 'privste'}
+        self.assertRaises(PlotlyError, py.plot, fig, **kwargs)
+
+    def test_plot_world_readable_sharing_conflict_1(self):
+
+        # Raise an error if world_readable=False but sharing='public'
+
+        fig = {
+            'data': [
+                {
+                    'x': [1, 2, 3],
+                    'y': [2, 1, 2]
+                }
+            ]
+        }
+
+        kwargs = {'filename': 'invalid-privacy-setting',
+                  'world_readable': False,
+                  'sharing': 'public'}
+        self.assertRaises(PlotlyError, py.plot, fig, **kwargs)
+
+    def test_plot_world_readable_sharing_conflict_2(self):
+
+        # Raise an error if world_readable=True but sharing='secret'
+
+        fig = {
+            'data': [
+                {
+                    'x': [1, 2, 3],
+                    'y': [2, 1, 2]
+                }
+            ]
+        }
+
+        kwargs = {'filename': 'invalid-privacy-setting',
+                  'world_readable': True,
+                  'sharing': 'secret'}
+        self.assertRaises(PlotlyError, py.plot, fig, **kwargs)
+
+    def test_plot_option_logic_only_world_readable_given(self):
+
+        # If sharing is not given and world_readable=False,
+        # sharing should be set to None
+
+        kwargs = {'filename': 'test',
+                  'auto_open': True,
+                  'fileopt': 'overwrite',
+                  'validate': True,
+                  'world_readable': False}
+
+        plot_option_logic = py._plot_option_logic(kwargs)
+
+        expected_plot_option_logic = {'filename': 'test',
+                                      'auto_open': True,
+                                      'fileopt': 'overwrite',
+                                      'validate': True,
+                                      'world_readable': False,
+                                      'sharing': None}
+        self.assertEqual(plot_option_logic, expected_plot_option_logic)
+
+    def test_plot_option_logic_only_sharing_given(self):
+
+        # If world_readable is not given and sharing ='private',
+        # world_readable should be set to False
+
+        kwargs = {'filename': 'test',
+                  'auto_open': True,
+                  'fileopt': 'overwrite',
+                  'validate': True,
+                  'sharing': 'private'}
+
+        plot_option_logic = py._plot_option_logic(kwargs)
+
+        expected_plot_option_logic = {'filename': 'test',
+                                      'auto_open': True,
+                                      'fileopt': 'overwrite',
+                                      'validate': True,
+                                      'world_readable': False,
+                                      'sharing': 'private'}
+        self.assertEqual(plot_option_logic, expected_plot_option_logic)
```

Comment:
@aneda - can we add a test that this actually works? i.e. 
- make a request to plotly with the share key
- confirm that the returned url has a share key in it
- make another get request to that url and make sure we get a 200 level response
- make another request to the url *without* the share key and make sure we get a 404 level response
it's technically testing stuff on the plotly server side, but it's good for this package and its tests to be aware that it is interfacing with plotly as expected.